src/App.tsx

```diff
@@ -1,8 +1,75 @@
-import React from 'react';
+import React, {
+  ChangeEvent, useState, FC, useMemo,
+} from 'react';
 import './App.css';
+import { PostList } from './components/PostList';
+import { getPostsWithUsersAndComments } from './components/getData/getData';
 
-const App = () => (
-  <h1>Dynamic list of posts</h1>
-);
+const App: FC = () => {
+  const [posts, setPosts] = useState<PostWithUserAndComment[]>([]);
```

Comment:
Зачем такой сложный интерфейс?
```suggestion
  const [posts, setPosts] = useState<Post[]>([]);
```